apps/cms.c

```diff
@@ -885,10 +892,22 @@ int cms_main(int argc, char **argv)
             goto end;
     }
 
-    in = bio_open_default(infile, 'r',
-                          binary_files ? FORMAT_BINARY : informat);
-    if (in == NULL)
-        goto end;
+    if (digesthex != NULL) {
+        if (infile != NULL
+            || operation != SMIME_SIGN
+            || !(flags & CMS_DETACHED)) {
+            BIO_printf(bio_err, "Invalid use of -digest\n");
```

Comment:
Point out what the concrete issue is, e.g,

Cannot use -digest when -in or -nodetach option is given

Reply:
Moreover, in case -sign is not given, better add a warning that -digest is ignored.

Reply:
Just to be sure, do you prefer a warning and ignoring the digest rather than an error for -digest without -sign?

Reply:
I think in cases like this where a needless option can safely ignored it's best just to give a warning and proceed.
This provides more flexibility and is what also other OpenSSL commands do.

Reply:
Adding "Warning: -digest is ignored for non-signing operation\n" inpired by the existing "Warning: -nodetach option is ignored for non-signing operation\n".